Kernel compilation issue 6.12.52

Hi,
might I ask if I'm missing something obvious compiling on Kernel 6.12.52? SRIOV for 6.12.47 compiled fine about a month ago.

If you need anything else please let me know, here is the output from my build toolchain:

```
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/sysfs_engines.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_bo.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_guc.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_lrc.o
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:132,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.h:467,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.c:8:
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h: In function 'ftrace_test_probe_xe_reg_rw':
./include/trace/trace_events.h:442:42: error: passing argument 1 of 'check_trace_callback_type_xe_reg_rw' from incompatible pointer type [-Wincompatible-pointer-types]
  442 |         check_trace_callback_type_##call(trace_event_raw_event_##template); \
      |                                          ^~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int) {aka void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)}
./include/trace/trace_events.h:46:9: note: in expansion of macro 'DEFINE_EVENT'
   46 |         DEFINE_EVENT(name, name, PARAMS(proto), PARAMS(args));
      |         ^~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h:344:1: note: in expansion of macro 'TRACE_EVENT'
  344 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
In file included from /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.h:12:
./include/linux/tracepoint.h:308:49: note: expected 'void (*)(void *, struct xe_mmio *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_mmio *, _Bool,  unsigned int,  long long unsigned int,  int)'} but argument is of type 'void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)'}
  308 |         check_trace_callback_type_##name(void (*cb)(data_proto))        \
      |                                          ~~~~~~~^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:453:9: note: in expansion of macro '__DECLARE_TRACE'
  453 |         __DECLARE_TRACE(name, PARAMS(proto), PARAMS(args),              \
      |         ^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:587:9: note: in expansion of macro 'DECLARE_TRACE'
  587 |         DECLARE_TRACE(name, PARAMS(proto), PARAMS(args))
      |         ^~~~~~~~~~~~~
/usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.h:350:1: note: in expansion of macro 'TRACE_EVENT'
  350 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:133:
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h: In function 'perf_test_probe_xe_reg_rw':
./include/trace/perf.h:67:42: error: passing argument 1 of 'check_trace_callback_type_xe_reg_rw' from incompatible pointer type [-Wincompatible-pointer-types]
   67 |         check_trace_callback_type_##call(perf_trace_##template);        \
      |                                          ^~~~~~~~~~~
      |                                          |
      |                                          void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int) {aka void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)}
./include/trace/trace_events.h:46:9: note: in expansion of macro 'DEFINE_EVENT'
   46 |         DEFINE_EVENT(name, name, PARAMS(proto), PARAMS(args));
      |         ^~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h:344:1: note: in expansion of macro 'TRACE_EVENT'
  344 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
./include/linux/tracepoint.h:308:49: note: expected 'void (*)(void *, struct xe_mmio *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_mmio *, _Bool,  unsigned int,  long long unsigned int,  int)'} but argument is of type 'void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)'}
  308 |         check_trace_callback_type_##name(void (*cb)(data_proto))        \
      |                                          ~~~~~~~^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:453:9: note: in expansion of macro '__DECLARE_TRACE'
  453 |         __DECLARE_TRACE(name, PARAMS(proto), PARAMS(args),              \
      |         ^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:587:9: note: in expansion of macro 'DECLARE_TRACE'
  587 |         DECLARE_TRACE(name, PARAMS(proto), PARAMS(args))
      |         ^~~~~~~~~~~~~
/usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.h:350:1: note: in expansion of macro 'TRACE_EVENT'
  350 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:134:
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h: In function 'bpf_test_probe_xe_reg_rw':
./include/trace/bpf_probe.h:64:42: error: passing argument 1 of 'check_trace_callback_type_xe_reg_rw' from incompatible pointer type [-Wincompatible-pointer-types]
   64 |         check_trace_callback_type_##call(__bpf_trace_##template);       \
      |                                          ^~~~~~~~~~~~
      |                                          |
      |                                          void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int) {aka void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)}
./include/trace/bpf_probe.h:100:9: note: in expansion of macro '__DEFINE_EVENT'
  100 |         __DEFINE_EVENT(template, call, PARAMS(proto), PARAMS(args), 0)
      |         ^~~~~~~~~~~~~~
./include/trace/trace_events.h:46:9: note: in expansion of macro 'DEFINE_EVENT'
   46 |         DEFINE_EVENT(name, name, PARAMS(proto), PARAMS(args));
      |         ^~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace.h:344:1: note: in expansion of macro 'TRACE_EVENT'
  344 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
./include/linux/tracepoint.h:308:49: note: expected 'void (*)(void *, struct xe_mmio *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_mmio *, _Bool,  unsigned int,  long long unsigned int,  int)'} but argument is of type 'void (*)(void *, struct xe_gt *, bool,  u32,  u64,  int)' {aka 'void (*)(void *, struct xe_gt *, _Bool,  unsigned int,  long long unsigned int,  int)'}
  308 |         check_trace_callback_type_##name(void (*cb)(data_proto))        \
      |                                          ~~~~~~~^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:453:9: note: in expansion of macro '__DECLARE_TRACE'
  453 |         __DECLARE_TRACE(name, PARAMS(proto), PARAMS(args),              \
      |         ^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:587:9: note: in expansion of macro 'DECLARE_TRACE'
  587 |         DECLARE_TRACE(name, PARAMS(proto), PARAMS(args))
      |         ^~~~~~~~~~~~~
/usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.h:350:1: note: in expansion of macro 'TRACE_EVENT'
  350 | TRACE_EVENT(xe_reg_rw,
      | ^~~~~~~~~~~
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/intel_ggtt_gmch.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/gen6_renderstate.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_ttm_stolen_mgr.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_ttm_sys_mgr.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_ttm_vram_mgr.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_tuning.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_uc.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/gen7_renderstate.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_uc_fw.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_vm.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/gen8_renderstate.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_vram.o
make[3]: *** [scripts/Makefile.build:229: /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace.o] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:132,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_bo.h:263,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_bo.c:8:
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h: In function 'do_trace_event_raw_event_xe_bo':
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:36:46: error: 'struct xe_bo' has no member named 'size'
   36 |                            __entry->size = bo->size;
      |                                              ^~
./include/trace/trace_events.h:405:11: note: in definition of macro '__DECLARE_EVENT_CLASS'
  405 |         { assign; }                                                     \
      |           ^~~~~~
./include/trace/trace_events.h:413:23: note: in expansion of macro 'PARAMS'
  413 |                       PARAMS(assign), PARAMS(print))                    \
      |                       ^~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:23:1: note: in expansion of macro 'DECLARE_EVENT_CLASS'
   23 | DECLARE_EVENT_CLASS(xe_bo,
      | ^~~~~~~~~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:34:21: note: in expansion of macro 'TP_fast_assign'
   34 |                     TP_fast_assign(
      |                     ^~~~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h: In function 'do_trace_event_raw_event_xe_bo_move':
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:66:38: error: 'struct xe_bo' has no member named 'size'
   66 |                    __entry->size = bo->size;
      |                                      ^~
./include/trace/trace_events.h:405:11: note: in definition of macro '__DECLARE_EVENT_CLASS'
  405 |         { assign; }                                                     \
      |           ^~~~~~
./include/trace/trace_events.h:413:23: note: in expansion of macro 'PARAMS'
  413 |                       PARAMS(assign), PARAMS(print))                    \
      |                       ^~~~~~
./include/trace/trace_events.h:40:9: note: in expansion of macro 'DECLARE_EVENT_CLASS'
   40 |         DECLARE_EVENT_CLASS(name,                              \
      |         ^~~~~~~~~~~~~~~~~~~
./include/trace/trace_events.h:44:30: note: in expansion of macro 'PARAMS'
   44 |                              PARAMS(assign),                   \
      |                              ^~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:51:1: note: in expansion of macro 'TRACE_EVENT'
   51 | TRACE_EVENT(xe_bo_move,
      | ^~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:64:13: note: in expansion of macro 'TP_fast_assign'
   64 |             TP_fast_assign(
      |             ^~~~~~~~~~~~~~
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:133:
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h: In function 'perf_trace_xe_bo':
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:36:46: error: 'struct xe_bo' has no member named 'size'
   36 |                            __entry->size = bo->size;
      |                                              ^~
./include/trace/perf.h:51:11: note: in definition of macro 'DECLARE_EVENT_CLASS'
   51 |         { assign; }                                                     \
      |           ^~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:34:21: note: in expansion of macro 'TP_fast_assign'
   34 |                     TP_fast_assign(
      |                     ^~~~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h: In function 'perf_trace_xe_bo_move':
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:66:38: error: 'struct xe_bo' has no member named 'size'
   66 |                    __entry->size = bo->size;
      |                                      ^~
./include/trace/perf.h:51:11: note: in definition of macro 'DECLARE_EVENT_CLASS'
   51 |         { assign; }                                                     \
      |           ^~~~~~
./include/trace/trace_events.h:44:30: note: in expansion of macro 'PARAMS'
   44 |                              PARAMS(assign),                   \
      |                              ^~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:51:1: note: in expansion of macro 'TRACE_EVENT'
   51 | TRACE_EVENT(xe_bo_move,
      | ^~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/xe/xe_trace_bo.h:64:13: note: in expansion of macro 'TP_fast_assign'
   64 |             TP_fast_assign(
      |             ^~~~~~~~~~~~~~
make[3]: *** [scripts/Makefile.build:229: /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe/xe_trace_bo.o] Error 1
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/gen9_renderstate.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_busy.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_clflush.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_dmabuf.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_create.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_context.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_domain.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_execbuffer.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_internal.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_lmem.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_mman.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_object.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_pages.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_phys.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_pm.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_region.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_shmem.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_shrinker.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_stolen.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_throttle.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_tiling.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_ttm.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_ttm_move.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_ttm_pm.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_userptr.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gem_wait.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gem/i915_gemfs.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_active.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_cmd_parser.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_deps.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_gem.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_gem_evict.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_gem_gtt.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_gem_ww.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_query.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_request.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_scheduler.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_trace_points.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_ttm_buddy_manager.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_vma.o
make[2]: *** [scripts/Makefile.build:478: /usr/src/i915-sriov-dkms/drivers/gpu/drm/xe] Error 2
make[2]: *** Waiting for unfinished jobs....
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_vma_resource.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_gsc_fw.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_gsc_proxy.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_gsc_uc.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_gsc_uc_debugfs.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_gsc_uc_heci_cmd_submit.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_ads.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_capture.o
In file included from /usr/src/i915-sriov-dkms/include/trace/define_trace.h:132,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_trace.h:726,
                 from /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_trace_points.c:13:
./include/trace/../../drivers/gpu/drm/i915/i915_trace.h: In function 'ftrace_test_probe_i915_reg_rw':
./include/trace/trace_events.h:442:9: error: implicit declaration of function 'check_trace_callback_type_i915_reg_rw'; did you mean 'check_trace_callback_type_i915_gem_evict'? [-Wimplicit-function-declaration]
  442 |         check_trace_callback_type_##call(trace_event_raw_event_##template); \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
./include/trace/trace_events.h:46:9: note: in expansion of macro 'DEFINE_EVENT'
   46 |         DEFINE_EVENT(name, name, PARAMS(proto), PARAMS(args));
      |         ^~~~~~~~~~~~
/usr/src/i915-sriov-dkms/include/trace/define_trace.h:32:9: note: in expansion of macro 'TRACE_EVENT'
   32 |         TRACE_EVENT(name,                                               \
      |         ^~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/i915/i915_trace.h:645:1: note: in expansion of macro 'TRACE_EVENT_CONDITION'
  645 | TRACE_EVENT_CONDITION(i915_reg_rw,
      | ^~~~~~~~~~~~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/i915/i915_trace.h: At top level:
./include/trace/trace_events.h:471:44: error: '__tracepoint_i915_reg_rw' undeclared here (not in a function); did you mean '__tracepoint_i915_gem_evict'?
  471 |                 .tp                     = &__tracepoint_##call,         \
      |                                            ^~~~~~~~~~~~~
./include/trace/trace_events.h:46:9: note: in expansion of macro 'DEFINE_EVENT'
   46 |         DEFINE_EVENT(name, name, PARAMS(proto), PARAMS(args));
      |         ^~~~~~~~~~~~
/usr/src/i915-sriov-dkms/include/trace/define_trace.h:32:9: note: in expansion of macro 'TRACE_EVENT'
   32 |         TRACE_EVENT(name,                                               \
      |         ^~~~~~~~~~~
./include/trace/../../drivers/gpu/drm/i915/i915_trace.h:645:1: note: in expansion of macro 'TRACE_EVENT_CONDITION'
  645 | TRACE_EVENT_CONDITION(i915_reg_rw,
      | ^~~~~~~~~~~~~~~~~~~~~
make[3]: *** [scripts/Makefile.build:229: /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/i915_trace_points.o] Error 1
make[3]: *** Waiting for unfinished jobs....
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_ct.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_debugfs.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_fw.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_log.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_hwconfig.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_log_debugfs.o
  CC [M]  /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915/gt/uc/intel_guc_rc.o
make[2]: *** [scripts/Makefile.build:478: /usr/src/i915-sriov-dkms/drivers/gpu/drm/i915] Error 2
make[1]: *** [/usr/src/linux-6.12.52-Unraid/Makefile:1945: /usr/src/i915-sriov-dkms] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,8 @@ LINUXINCLUDE := \
 
 subdir-ccflags-y += \
 	-DDKMS_MODULE_VERSION='$(DKMS_MODULE_VERSION)' \
-	-DDKMS_MODULE_ORIGIN_KERNEL='$(DKMS_MODULE_ORIGIN_KERNEL)'
+	-DDKMS_MODULE_ORIGIN_KERNEL='$(DKMS_MODULE_ORIGIN_KERNEL)' \
+	-DDKMS_MODULE_SOURCE_DIR='$(abspath $(src))'
 
 obj-m += compat/
 obj-m += drivers/gpu/drm/i915/

--- a/drivers/gpu/drm/i915/display/intel_display_trace.h
+++ b/drivers/gpu/drm/i915/display/intel_display_trace.h
@@ -853,6 +853,6 @@ TRACE_EVENT(intel_frontbuffer_flush,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/i915/display
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/i915/display)
 #define TRACE_INCLUDE_FILE intel_display_trace
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/i915/gvt/trace.h
+++ b/drivers/gpu/drm/i915/gvt/trace.h
@@ -378,6 +378,6 @@ TRACE_EVENT(render_mmio,
 /* This part must be out of protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/i915/gvt
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/i915/gvt)
 #define TRACE_INCLUDE_FILE trace
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/i915/i915_trace.h
+++ b/drivers/gpu/drm/i915/i915_trace.h
@@ -721,6 +721,6 @@ DEFINE_EVENT(i915_context, i915_context_free,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/i915
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/i915)
 #define TRACE_INCLUDE_FILE i915_trace
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/i915/intel_uncore_trace.h
+++ b/drivers/gpu/drm/i915/intel_uncore_trace.h
@@ -44,6 +44,6 @@ TRACE_EVENT_CONDITION(i915_reg_rw,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/i915
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/i915)
 #define TRACE_INCLUDE_FILE intel_uncore_trace
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/xe/xe_trace.h
+++ b/drivers/gpu/drm/xe/xe_trace.h
@@ -462,6 +462,6 @@ TRACE_EVENT(xe_eu_stall_data_read,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/xe
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/xe)
 #define TRACE_INCLUDE_FILE xe_trace
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/xe/xe_trace_bo.h
+++ b/drivers/gpu/drm/xe/xe_trace_bo.h
@@ -258,6 +258,6 @@ DEFINE_EVENT(xe_vm, xe_vm_ops_fail,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/xe
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/xe)
 #define TRACE_INCLUDE_FILE xe_trace_bo
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/xe/xe_trace_guc.h
+++ b/drivers/gpu/drm/xe/xe_trace_guc.h
@@ -154,6 +154,6 @@ TRACE_EVENT(xe_guc_engine_activity,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/xe
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/xe)
 #define TRACE_INCLUDE_FILE xe_trace_guc
 #include <trace/define_trace.h>

--- a/drivers/gpu/drm/xe/xe_trace_lrc.h
+++ b/drivers/gpu/drm/xe/xe_trace_lrc.h
@@ -47,6 +47,6 @@ TRACE_EVENT(xe_lrc_update_timestamp,
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
 #undef TRACE_INCLUDE_FILE
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/xe
+#define TRACE_INCLUDE_PATH MODULE_ABS_PATH(drivers/gpu/drm/xe)
 #define TRACE_INCLUDE_FILE xe_trace_lrc
 #include <trace/define_trace.h>

--- a/include/config.h
+++ b/include/config.h
@@ -2,6 +2,8 @@
 
 #include "backport.h"
 
+#define MODULE_ABS_PATH(path) DKMS_MODULE_SOURCE_DIR/path
+
 #ifdef CONFIG_DRM_XE_PAGEMAP
 #undef CONFIG_DRM_XE_PAGEMAP
 #endif

